media_kit_video/lib/src/video_controller/web_video_controller/real.dart

```diff
@@ -0,0 +1,109 @@
+/// This file is a part of media_kit (https://github.com/alexmercerind/media_kit).
+///
+/// Copyright © 2021 & onwards, Hitesh Kumar Saini <saini123hitesh@gmail.com>.
+/// All rights reserved.
+/// Use of this source code is governed by MIT license that can be found in the LICENSE file.
+// ignore_for_file: avoid_web_libraries_in_flutter
+import 'dart:ui';
+import 'dart:async';
+import 'dart:js' as js;
+import 'dart:html' as html;
+import 'package:flutter/foundation.dart';
+import 'package:media_kit/media_kit.dart';
+
+import 'package:media_kit_video/src/video_controller/platform_video_controller.dart';
+
+/// {@template web_video_controller}
+///
+/// WebVideoController
+/// ------------------
+///
+/// The [WebVideoController] implementation based on HTML & JavaScript used on web.
+///
+/// {@endtemplate}
+class WebVideoController extends PlatformVideoController {
+  /// Whether [WebVideoController] is supported on the current platform or not.
+  static const bool supported = kIsWeb;
+
+  /// {@macro web_video_controller}
+  WebVideoController._(
+    super.player,
+    super.width,
+    super.height,
+    super.enableHardwareAcceleration,
+  );
+
+  /// {@macro web_video_controller}
+  static Future<PlatformVideoController> create(Player player) async {
+    // Retrieve the native handle of the [Player].
+    final handle = await player.handle;
+
+    final controller = WebVideoController._(
+      player,
+      null,
+      null,
+      true,
+    );
+    // Register [_dispose] for execution upon [Player.dispose].
+    player.platform?.release.add(controller._dispose);
+
+    // Retrieve the [html.VideoElement] instance from [js.context].
+    controller._element = js.context[_kInstances][handle];
+    // Register the [html.VideoElement] as platform view.
+    platformViewRegistry.registerViewFactory(
```

Comment:
I have this error with `flutter 3.10.1-stable`, but does not prevent the build from succeeding on non-web platforms:

```shell
$ cd media_kit_video
$ flutter analyze
Analyzing media_kit_video...                                            

  error • Undefined name 'platformViewRegistry' • lib/src/video_controller/web_video_controller/real.dart:53:5 • undefined_identifier

1 issue found. (ran in 2.5s)
```

Reply:
Was this resolved for you?

Reply:
It should be good now. One more conditional import.